src/gc/gc.cpp

```diff
@@ -26685,7 +26683,7 @@ void gc_heap::background_grow_c_mark_list()
         assert (new_c_mark_list);
         memcpy (new_c_mark_list, c_mark_list, c_mark_list_length*sizeof(uint8_t*));
         c_mark_list_length = c_mark_list_length*2;
-        delete c_mark_list;
+        delete [] c_mark_list;
         c_mark_list = new_c_mark_list;
```

Comment:
@Maoni0 new_c_mark_list was always created with new [].  So the initial version of c_mark_list was created with new.  Subsequent version were created with new [] making it impossible to call the correct delete on destruction to comply with C++ standard.